File: app/src/main/java/com/yanrou/dawnisland/MyViewPagerAdapter.java
package com.yanrou.dawnisland;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class MyViewPagerAdapter extends FragmentPagerAdapter {
    private List<Fragment> fragments = new ArrayList<>();

    public MyViewPagerAdapter(@NonNull FragmentManager fm, int behavior) {
        super(fm, behavior);
    }


    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    public void addFragment(Fragment fragment) {
        fragments.add(fragment);
    }

    public void addFragments(List<Fragment> fragments) {
        this.fragments.addAll(fragments);
    }
}

File: app/src/main/java/com/yanrou/dawnisland/MyViewPagerAdapter.kt
package com.yanrou.dawnisland

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.yanrou.dawnisland.feed.FeedFragment
import com.yanrou.dawnisland.serieslist.SeriesFragment
import com.yanrou.dawnisland.trend.TrendFragment

class MyViewPagerAdapter(fm: FragmentManager, behavior: Int) : FragmentPagerAdapter(fm, behavior) {
    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> SeriesFragment()
            1 -> TrendFragment()
            2 -> FeedFragment()
            else -> throw IllegalStateException("索引越界")
        }
    }

    override fun getCount() = 3


}